$OpenBSD: patch-Makefile,v 1.45 2020/09/12 19:35:37 sthen Exp $

- don't force compiler flags
- don't unconditionally set -ftrampolines, it is gcc-only.
(Blocksruntime isn't working correctly on OpenBSD anyway so clang
is not yet an option, but patched away here to allow for easier
testing, and added-back in the port Makefile).

Index: Makefile
--- Makefile.orig
+++ Makefile
@@ -138,7 +138,7 @@ ASTTOPDIR:=$(subst $(space),\$(space),$(CURDIR))
 OVERWRITE=y
 
 # Include debug and macro symbols in the executables (-g) and profiling info (-pg)
-DEBUG=-g3
+#DEBUG=-g3
 
 # Asterisk.conf is located in ASTETCDIR or by using the -C flag
 # when starting Asterisk
@@ -187,7 +187,9 @@ OPTIONS=
 
 ifeq ($(findstring -save-temps,$(_ASTCFLAGS) $(ASTCFLAGS)),)
   ifeq ($(findstring -pipe,$(_ASTCFLAGS) $(ASTCFLAGS)),)
+    ifneq ($(OSARCH),OpenBSD)
     _ASTCFLAGS+=-pipe
+    endif
   endif
 endif
 
@@ -231,7 +233,7 @@ ifeq ($(OSARCH),NetBSD)
 endif
 
 ifeq ($(OSARCH),OpenBSD)
-  _ASTCFLAGS+=-pthread -ftrampolines
+  _ASTCFLAGS+=-pthread
 endif
 
 ifeq ($(OSARCH),linux-uclibc)
